libraries/Microsoft.Bot.Builder.Dialogs.Adaptive/DialogManager.cs

```diff
@@ -15,23 +19,23 @@ namespace Microsoft.Bot.Builder.Dialogs
     /// </summary>
     public class DialogManager
     {
-        private const string LASTACCESS = "_lastAccess";
-        private string rootDialogId;
-        private string dialogStateProperty;
+        private const string LastAccess = "_lastAccess";
+        private string _rootDialogId;
```

Comment:
>_rootDialogId [](start = 23, length = 13)

This is not the coding style we have used throughout the sdk.  Why are you applying ithere?

Reply:
That's the code style we use in anything that is not adaptive (see https://github.com/microsoft/botbuilder-dotnet/blob/master/libraries/Microsoft.Bot.Builder/BotFrameworkAdapter.cs and many other classes in BotBuilder and other libraries).

@mrivera-ms & @cleemullins can you guys chime in and clarify what should we use?

Reply:
So on the JS side of things I often have to use _ prefixes when defining the internal member variable for a public property but otherwise I avoid _

---
In reply to: [405685793](https://github.com/microsoft/botbuilder-dotnet/pull/3709#discussion_r405685793) [](ancestors = 405685793)

Reply:
In C#, the most common approach is to use _ to prefix private fields in a class (this also saves you from having to use this to disambiguate local function variables from class fields).

You can see other examples if you browse the ASP.Net core classes: https://github.com/dotnet/aspnetcore/blob/master/src/Hosting/WindowsServices/src/WebHostService.cs

Reply:
this. is always accurate and always conveys exactly what it means, _ is not
var _foo = "" 
does NOTHING to tell you about the scope.

I have coded all of the bot framework with out _, why did we start now?


---
In reply to: [405839791](https://github.com/microsoft/botbuilder-dotnet/pull/3709#discussion_r405839791) [](ancestors = 405839791)

Reply:
I don't think this is something I made up, this has been going on for a while, the private fields without an _ are more an anomaly in the framework than the common norm. Most of the team seems to have adopted this approach and we use the same convention for samples. Only the new adaptive libraries seem to be using a different convention.

Reply:
I checked the code base and it seems there is no adopted convention in the usage of underscore prefixes for private fields. In fact, there slightly a higher number of underscores across all libraries. We should, however, adopt one and standardize.

Reply:
I talked to @cleemullins, the standard for the C# SDK is _.